addons/account/models/account_payment_method.py

```diff
@@ -92,6 +92,7 @@ class AccountPaymentMethodLine(models.Model):
     _name = "account.payment.method.line"
     _description = "Payment Methods"
     _order = 'sequence, id'
+    _check_company_domain = models.check_company_domain_parent_of
```

Comment:
Why do you need to change this? :thinking:

Reply:
It's to allow `check_company` on a `account.payment.method.line` type field to include the branches in the check.

Reply:
But do we want that?
The commit message doesn't say anything about it, and I'm not sure it is right functionally

Reply:
JulienAlardot thought that the company condition has been removed from the domain of the field to include the branches.

Here's how it was in 16.3:

https://github.com/odoo/odoo/blob/3dee259097dd2600ab8457e5737506c9405f92e5/addons/hr_expense/models/res_company.py#L16-L21

And here's how it is in 16.4:

https://github.com/odoo/odoo/blob/c6c0f6ea677f5cb5935d6a3766bca5990539d17a/addons/hr_expense/models/res_company.py#L22-L27

There's a point in the solution section about the override of `_check_company_domain`.
But I don't know if it we really want it.
Should I remove it? :thinking:

Reply:
No, it has been remove in order to be consistent with whatever `_check_company_domain` is.
By default, we should never have an explicit domain on the companies since the branches exist. But the fact that it was removed doesn't mean that the domain changed.

Reply:
Ah ok. That's why.
I will remove the override then.

Thanks. :)